clash-lib/src/Clash/Backend/VHDL.hs

```diff
@@ -184,7 +187,9 @@ instance Backend VHDLState where
     Void {} -> pure PrimitiveType
     KnownDomain {} -> pure PrimitiveType
 
-  hdlType Internal      (filterTransparent -> ty) = sizedQualTyName ty
+  hdlType Internal      (filterTransparent -> ty) = do
+    pkgCtx <- Ap (use tyPkgCtx)
+    if pkgCtx then sizedTyName ty else sizedQualTyName ty
```

Comment:
Didn't you make `qualTyName` (used by `sizedQualTyName`) smart enough to not do the qualification when `tyPkgCtx` is set?

Reply:
Right, yes. Or rather, `sizedQualTyName` calls `qualTyName` which I indeed changed